common/src/main/java/io/netty/util/concurrent/DefaultPromise.java

```diff
@@ -68,17 +72,34 @@
      * It is preferable to use {@link EventExecutor#newPromise()} to create a new promise
      *
      * @param executor
-     *        the {@link EventExecutor} which is used to notify the promise once it is complete.
+     *        The {@link EventExecutor} which is used to notify the promise once it is complete.
      *        It is assumed this executor will protect against {@link StackOverflowError} exceptions.
      *        The executor may be used to avoid {@link StackOverflowError} by executing a {@link Runnable} if the stack
      *        depth exceeds a threshold.
-     *
      */
     public DefaultPromise(EventExecutor executor) {
         this.executor = requireNonNull(executor, "executor");
         stage = new DefaultFutureCompletionStage<>(this);
     }
 
+    /**
+     * Creates a completed promise.
+     */
+    DefaultPromise(EventExecutor executor, Object result) {
```

Comment:
I like getting rid of FailedFuture and SucceededFuture. There's no real benefit to the specialized classes and they may prevent some JIT optimizations.